bascule/key/cache.go

```diff
@@ -164,14 +189,20 @@ func (cache *multiCache) UpdateKeys(ctx context.Context) (count int, errors []er
 			newCount := 0
 			newPairs := make(map[string]Pair, len(existingPairs))
 			for keyID, oldPair := range existingPairs {
-				select {
-				case <-ctx.Done():
+				if isContextDone(ctx) {
+					count = 0
+					errors = append(errors, ErrOperationTimedOut)
+					return
+				}
+				newPair, err := cache.delegate.ResolveKey(ctx, keyID)
+
+				if isContextDone(ctx) {
```

Comment:
You don't need this second check.  The delegate is responsible for returning an error if a context was canceled during its execution.